internal/workerutil/dbworker/store/store.go

```diff
@@ -371,6 +403,32 @@ WHERE
 	{state} IN (%s)
 `
 
+func (s *store[T]) Exists(ctx context.Context, statesBitset RecordState) (_ bool, err error) {
+	ctx, _, endObservation := s.operations.queuedCount.With(ctx, &err, observation.Args{})
```

Comment:
we should track this separately 
```suggestion
	ctx, _, endObservation := s.operations.exists.With(ctx, &err, observation.Args{})
```